src/optimizer/rule/IndexScanRule.cpp

```diff
@@ -49,7 +49,9 @@ StatusOr<OptRule::TransformResult> IndexScanRule::transform(OptContext* ctx,
         NG_RETURN_IF_ERROR(createIndexQueryCtx(iqctx, kind, items, qctx, groupNode));
     }
 
-    auto newIN = static_cast<const IndexScan*>(groupNode->node())->clone(qctx);
+    auto* oldIN = groupNode->node();
+    DCHECK_EQ(oldIN->kind(), graph::PlanNode::Kind::kIndexScan);
+    auto* newIN = static_cast<IndexScan*>(oldIN->clone());
```

Comment:
Why remove the `const` modifier.

Reply:
No changes, oldIN is still const.

Reply:
The `const` in static_cast

Reply:
newIN is not const, It gonna be modified.